### ISSUE TITLE
valid_docs incorrectly expects return documentation when method definition includes inline closure argument

### ISSUE THREAD

**Issue by chadmoone** [realm/SwiftLint#656]
The following will incorrectly trigger an `invalid_docs` warning:

```swift
 /**
Asynchronous method which does something.

- parameter foo: The foo with which to do the thing.
- parameter completion: A closure called upon completion.
*/
public func doSomething(foo: String, completion: () -> ()) {
    // Implementation
}
```

Adding a `- returns:` declaration will silence the error:
```swift
 /**
Asynchronous method which does something.

- parameter foo: The foo with which to do the thing.
- parameter completion: A closure called upon completion.

- returns: blahblahblah
*/
public func doSomething(foo: String, completion: () -> ()) {
    // Implementation
}
```

This seems like a pretty common occurrence.

**Comment by sofili** [comment]
same for ///

**Comment by cdzombak** [comment]
does this duplicate #580?

**Comment by chadmoone** [comment]
It looks like this is the same issue.  Changing the closure return type to `Void` will prevent the warning here, and adding an erroneous return documentation on the function will prevent the warning in #580.

Verified with 0.11.1

**Comment by jpsim** [comment]
going to close this as a duplicate of #580
